### ISSUE TITLE
feat: Label smoothing in CrossEntropyLoss

### ISSUE THREAD

**Issue by Samyssmile** [Samyssmile/edux#201]
## Summary

`CrossEntropyLoss` uses hard one-hot targets. Label smoothing is a standard regularization technique that prevents the model from becoming overconfident, improving generalization.

**Estimated impact**: ~0.5–1% Top-1 accuracy improvement on ImageNet.

## Current State

```java
// CrossEntropyLoss.java
public LossResult compute(Tensor predictions, Tensor targets) {
    var probs = computeOps.softmax(predictions);
    float loss = 0f;
    for (int i = 0; i < totalElements; i++) {
        loss -= targetData[i] * (float) Math.log(Math.max(probData[i], 1e-7f));
    }
    loss /= batchSize;
    
    // Gradient: (softmax - target) / batch
    var gradient = probs.sub(targets).mul(1.0f / batchSize);
    return new LossResult(loss, gradient);
}
```

Targets are expected as hard one-hot: `[0, 0, 1, 0, ..., 0]`.

## Proposed Solution

Add optional label smoothing parameter:

```java
public final class CrossEntropyLoss implements Loss {
    private final float smoothing;  // default 0.0 (no smoothing)

    public CrossEntropyLoss() { this(0f); }

    /**
     * @param smoothing label smoothing factor in [0, 1). 
     *                  Typical value: 0.1 for ImageNet.
     *                  0.0 = hard labels (default, backward compatible).
     */
    public CrossEntropyLoss(float smoothing) {
        if (smoothing < 0 || smoothing >= 1) {
            throw new IllegalArgumentException("smoothing must be in [0, 1)");
        }
        this.smoothing = smoothing;
    }

    @Override
    public LossResult compute(Tensor predictions, Tensor targets) {
        var probs = computeOps.softmax(predictions);
        int numClasses = predictions.shape().get(predictions.shape().rank() - 1);

        // Apply label smoothing to targets
        // smoothed = (1 - ε) * one_hot + ε / numClasses
        Tensor smoothedTargets = targets;
        if (smoothing > 0f) {
            smoothedTargets = targets.mul(1f - smoothing)
                                     .addInPlace(smoothing / numClasses);
        }

        // Loss: -sum(smoothed_target * log(prob)) / batch
        // Gradient: (prob - smoothed_target) / batch
        // ... (same formula, just with smoothed targets)
    }
}
```

### Usage
```java
var loss = new CrossEntropyLoss(0.1f);  // ε = 0.1, standard for ImageNet

var trainer = new Trainer(model, loss, optimizer);
```

### Mathematical Effect

Without smoothing (ε = 0):
```
target = [0, 0, 1, 0, 0]  → model forced to predict exactly 1.0
```

With smoothing (ε = 0.1, 5 classes):
```
target = [0.02, 0.02, 0.92, 0.02, 0.02]  → softer target distribution
```

This prevents the model from pushing logits to infinity, which:
- Reduces overfitting
- Improves calibration
- Acts as implicit regularization

## Acceptance Criteria

- [ ] `CrossEntropyLoss()` (no-arg) preserves exact current behavior
- [ ] `CrossEntropyLoss(float smoothing)` applies label smoothing
- [ ] Smoothing validated in range [0, 1)
- [ ] Gradient computation correctly uses smoothed targets
- [ ] Loss value correctly computed with smoothed targets
- [ ] No extra tensor allocation when smoothing=0
- [ ] Unit test: smoothing=0 matches current behavior exactly
- [ ] Unit test: smoothing=0.1 produces correct smoothed target distribution
- [ ] Unit test: gradient correctness with smoothed labels

## References

- [Rethinking the Inception Architecture (Szegedy et al., 2016)](https://arxiv.org/abs/1512.00567) — Section 7 introduces label smoothing
- [When Does Label Smoothing Help? (Müller et al., 2019)](https://arxiv.org/abs/1906.02629)

**Comment by Samyssmile** [comment]
Implemented in 660f01a — label smoothing support in CrossEntropyLoss with full test coverage.
